lightning/src/ln/outbound_payment.rs

```diff
@@ -883,11 +886,11 @@ impl OutboundPayments {
 		}
 
 		let mut route_params = RouteParameters::from_payment_params_and_value(
-			PaymentParameters::from_bolt12_invoice(&invoice), invoice.amount_msats()
+			PaymentParameters::from_bolt12_invoice(&invoice).with_user_config(params_config), invoice.amount_msats()
 		);
-		if let Some(max_fee_msat) = max_total_routing_fee_msat {
-			route_params.max_total_routing_fee_msat = Some(max_fee_msat);
-		}
+
+		params_config.max_total_routing_fee_msat.map(|fee_msat| route_params.max_total_routing_fee_msat = Some(fee_msat));
```

Comment:
Prefer `if let` instead of putting a side effect in `map`.

Reply:
Done! [pr3342.06](https://github.com/shaavan/rust-lightning/commits/pr3342.06)
The code looks much more readable now!

Thanks for the pointer!